src/core/tritonserver.cc

```diff
@@ -1093,69 +1092,72 @@ TRITONSERVER_ServerModelMetadata(
       lserver->ModelReadyVersions(model_name, &ready_versions));
 
   rapidjson::Document metadata;
+  auto& allocator = metadata.GetAllocator();
   metadata.SetObject();
   // Just store string reference in JSON object since it will be serialized to
   // another buffer.
-  metadata.AddMember(
-      "name", rapidjson::StringRef(model_name), metadata.GetAllocator());
+  metadata.AddMember("name", rapidjson::StringRef(model_name), allocator);
 
   rapidjson::Value versions(rapidjson::kArrayType);
-  for (const auto v : ready_versions) {
-    auto version_str = std::to_string(v);
-    versions.PushBack(
-        rapidjson::Value(version_str.c_str(), version_str.size()),
-        metadata.GetAllocator());
+  if (model_version_int != -1) {
+    auto version_str = std::to_string(model_version_int);
```

Comment:
Need to check if it is one of the ready version...

Reply:
My bad, I think `GetInferenceBackend()` will return error in that case. The version should be ready if control reaches here.